### ISSUE TITLE
Offset tracking and committing

### ISSUE THREAD

**Issue by jaley** [ably/kafka-connect-ably#121]
We need to keep track of `SinkRecord` offsets somewhere centrally - perhaps create a new injectable component with an interface a bit like this?

```java
interface OffsetRegistry {
  void update(TopicPartition partition, long offset);
  long latestOffset(TopicPartition partition);
}
```

The `flush()` implementation would then make use of this component to commit progress back to Kafka when needed.

**Comment by subkanthi** [comment]
The idea is to store the offset information by partition in a service class
And return that information in the `preCommit` function.
```
    @Override
    public Map<TopicPartition, OffsetAndMetadata> preCommit(
            Map<TopicPartition, OffsetAndMetadata> offsets) throws RetriableException {

        return offsets;
    }
```

![Screenshot from 2023-06-17 12-52-28](https://github.com/ably/kafka-connect-ably/assets/17434846/7cbef428-6374-4457-9599-d78b7207cdf7)

**Comment by jaley** [comment]
Closed in #130
